examples/window/custom_user_event.rs

```diff
@@ -0,0 +1,114 @@
+//! Shows how to create a custom event that can be handled by `winit`'s event loop.
+
+use bevy::prelude::*;
+use bevy::winit::{EventLoopProxy, WakeUp, WinitPlugin};
+use std::fmt::Formatter;
+
+#[derive(Default, Debug, Event)]
+enum CustomEvent {
+    #[default]
+    WakeUp,
+    Key(char),
+}
+
+impl std::fmt::Display for CustomEvent {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        match self {
+            Self::WakeUp => write!(f, "Wake up"),
+            Self::Key(ch) => write!(f, "Key: {ch}"),
+        }
+    }
+}
+
+fn main() {
+    let winit_plugin = WinitPlugin::<CustomEvent>::default();
+
+    App::new()
+        .add_plugins(
+            DefaultPlugins
+                .build()
+                // Only one event type can be handled at once
+                // so we must disable the default event type
+                .disable::<WinitPlugin<WakeUp>>()
+                .add(winit_plugin),
+        )
+        .add_systems(
+            Startup,
+            (
+                setup,
+                #[cfg(target_arch = "wasm32")]
+                wasm::expose_event_loop_proxy,
+            ),
+        )
+        .add_systems(Update, (send_event, handle_event))
+        .run();
+}
+
+fn setup(mut commands: Commands) {
+    commands.spawn(Camera2dBundle::default());
+}
+
+fn send_event(
+    input: Res<ButtonInput<KeyCode>>,
+    event_loop_proxy: NonSend<EventLoopProxy<CustomEvent>>,
+) {
+    if input.just_pressed(KeyCode::Space) {
+        let _ = event_loop_proxy.send_event(CustomEvent::WakeUp);
+    }
+    if input.just_pressed(KeyCode::KeyE) {
+        let _ = event_loop_proxy.send_event(CustomEvent::Key('e'));
+    }
+}
+
+fn handle_event(mut events: EventReader<CustomEvent>) {
+    for evt in events.read() {
+        info!("Received event: {evt:?}");
+    }
+}
+
+#[cfg(target_arch = "wasm32")]
+pub(crate) mod wasm {
```

Comment:
could you add more comments on the wasm part?

Reply:
Done